elide-datastore/elide-datastore-aggregation/src/main/java/com/yahoo/elide/datastores/aggregation/queryengines/sql/SQLQueryEngine.java

```diff
@@ -356,9 +356,18 @@ private String extractOrderBy(Class<?> entityClass, Map<Path, Sorting.SortOrder>
 
                     Path.PathElement last = path.lastElement().get();
 
-                    return FilterPredicate.getTypeAlias(last.getType())
+                    String orderByType = FilterPredicate.getTypeAlias(last.getType())
                             + "."
-                            + getColumnName(entityClass, last.getFieldName())
+                            + getColumnName(entityClass, last.getFieldName());
+
+                    List<Metric> metricField = query.getMetrics().keySet().stream()
+                            .filter(m -> m.getName().equals(last.getFieldName())).collect(Collectors.toList());
```

Comment:
It isn't safe to just check the field name of the last element of the path.  Field names could be duplicated in different entities.  We really need to check here that the path is of size one (a single path element) and the field name matches a metric in the query.

I also think it would be good to promote the function to search for a metric in the query as a default method in the Query itself.